File: robok/robok-lang/src/main/java/robok/lang/primitive/Primitives.java
package robok.lang.primitive;

/*
  This class is responsible for getting whether the class is primitive or not.
*/

public class Primitives {
	private static final String[] primitives = new String[]{
		"byte",
		"short",
		"int",
		"long",
		"float",
		"double",
		"char",
		"boolean"
	};
	
	public Primitives(){
		
	}
	
	public boolean getPrimitiveType(String code){
		boolean isPrimitive = false;
		
		for(int i = 0; i < primitives.length; i++){
			if(code.equals(primitives[i])){
				isPrimitive = true;
				break;
			}
		}
		
		return isPrimitive;
	}
}

File: robok/robok-lang/src/main/java/org/robok/lang/primitive/Primitives.kt
package org.robok.lang.primitive

/*
 *  This file is part of Robok © 2024.
 *
 *  Robok is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Robok is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *   along with Robok.  If not, see <https://www.gnu.org/licenses/>.
 */

/*
  This class is responsible for getting whether the class is primitive or not.
*/

class Primitives {

    private val primitives = arrayOf(
        "byte", "short", "int", "long", "float", "double", "char", "boolean"
    )

    fun getPrimitiveType(code: String): Boolean {
        return primitives.contains(code)
    }
}